modules/common/hardware/lenovo-x1/definitions/default.nix

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+{
+  generation,
+  lib,
+}: let
+  hwDefinition = import (./. + "/x1-${generation}.nix");
+in {
+  inherit (hwDefinition) mouse;
+  inherit (hwDefinition) touchpad;
+  inherit (hwDefinition) disks;
+  inherit (hwDefinition) network;
+  inherit (hwDefinition) gpu;
+
+  # Notes:
+  #   1. This assembles udev rules for different hw configurations (i.e., different mice/touchpads) by adding
+  #      all of them to the configuration. This was chosen for simplicity to not have to provide hw identifier at build,
+  #      but is not ideal and should be changed.
+  #   2. USB camera "passthrough" is handled by qemu and thus available on host. If peripheral VM is implemented,
+  #      the entire host controller should be passthrough'd using the PCI bus (14.0). In x1, bluetooth and fingerprint
+  #      reader are on this bus.
+  udevRules = let
+    mapMouseRules =
+      builtins.map (d: ''        SUBSYSTEM=="input", ATTRS{name}=="${d}", KERNEL=="event*", GROUP="kvm", SYMLINK+="mouse"
```

Comment:
I know this is a refactor commit, but interestingly now as code gets more clear to read, some things become more visible.

I was thinking could there be something better than the `name` attribute to match to, like combination of `idVendor` and `idProduct`? I'm just thinking the name probably gets read from some database, which might change in the future, but at least the USB vendor id and product id should stay same.

Reply:
but this can also be fixed in some future commit